lib/tapioca/generators/base_generator.rb

```diff
@@ -0,0 +1,110 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "active_support/all"
+
+module Tapioca
+  module Generators
+    class BaseGenerator
+      extend(T::Sig)
+      extend(T::Helpers)
+
+      abstract!
+
+      sig { returns(Config) }
+      attr_reader :config
+
+      sig { params(config: Config).void }
+      def initialize(config)
+        @config = config
+        @bundle = T.let(nil, T.nilable(Gemfile))
+        @compiler = T.let(nil, T.nilable(Compilers::SymbolTableCompiler))
+        @loader = T.let(nil, T.nilable(Loader))
```

Comment:
Is this going to be the base class of all generators? These attributes seems strongly coupled with gem RBIs.